crates/gitbutler-testsupport/src/test_project.rs

```diff
@@ -214,64 +214,60 @@ impl TestProject {
     }
 
     /// works like if we'd open and merge a PR on github. does not update local.
-    pub fn merge(&self, branch_name: &Refname) {
+    pub fn merge(&self, branch_name: &Refname) -> anyhow::Result<()> {
         let branch_name: Refname = match branch_name {
-            Refname::Local(local) => format!("refs/heads/{}", local.branch()).parse().unwrap(),
-            Refname::Remote(remote) => format!("refs/heads/{}", remote.branch()).parse().unwrap(),
-            _ => "INVALID".parse().unwrap(), // todo
+            Refname::Local(local) => format!("refs/heads/{}", local.branch()).parse()?,
+            Refname::Remote(remote) => format!("refs/heads/{}", remote.branch()).parse()?,
+            _ => "INVALID".parse()?, // todo
         };
         let branch = self
             .remote_repository
-            .maybe_find_branch_by_refname(&branch_name)
-            .unwrap();
-        let branch_commit = branch.as_ref().unwrap().get().peel_to_commit().unwrap();
+            .maybe_find_branch_by_refname(&branch_name)?
+            .expect("branch exists");
```

Comment:
Why do we want to expect this one?

Reply:
oh... this is a test... why do we want to return a result?

Reply:
Large amounts of unwraps make code much less readable. My take on it is to…

• …not use them at all in production code
* …not use them at all in utility code for testing
* …use them while implementing a test just to…
* …finally remove them from the test in favor of `?` once the crate stabilises

As a crate stabilizes tests are unlikely to fail outside of assertions, so more readable code is preferable. Should a test start failing by returning an error, replacing `?` with unwraps temporarily is easy enough as well.

Reply:
Interesting. My take was that, using the results can obscure the stacktrace, and when I'm writing a test, if something other than what I'm testing is failing, it would be nice if it was painfully clear where that testing utility was going wrong.

It is however, undeniably more concise.

Reply:
> My take was that, using the results can obscure the stacktrace, and when I'm writing a test, if something other than what I'm testing is failing, it would be nice if it was painfully clear where that testing utility was going wrong.

And I am thinking the same. That's why I start tests with `.unwrap()` but migrate them towards `?` once the crate stabilises. That way the code is always most suited for the development stage one is in.

Reply:
Sounds reasonable